.github/buildomat/jobs/package.sh

```diff
@@ -42,39 +49,77 @@ files=(
 
 ptime -m tar cvzf /work/package.tar.gz "${files[@]}"
 
-# Assemble global zone files in a temporary directory.
-if ! tmp=$(mktemp -d); then
+#
+# Global Zone files for Host OS
+#
+
+if ! tmp_gz=$(mktemp -d); then
   exit 1
 fi
-trap 'cd /; rm -rf "$tmp"' EXIT
+trap 'cd /; rm -rf "$tmp_gz"' EXIT
 
 # Header file, identifying this is intended to be layered in the global zone.
 # Within the ramdisk, this means that all files under "root/foo" should appear
 # in the global zone as "/foo".
-echo '{"v":"1","t":"layer"}' > "$tmp/oxide.json"
+echo '{"v":"1","t":"layer"}' > "$tmp_gz/oxide.json"
 
 # Extract the sled-agent tarball for re-packaging into the layered GZ archive.
-pkg_dir="$tmp/root/opt/oxide/sled-agent"
+pkg_dir="$tmp_gz/root/opt/oxide/sled-agent"
 mkdir -p "$pkg_dir"
 cd "$pkg_dir"
 tar -xvfz "$tarball_src_dir/omicron-sled-agent.tar"
 # Ensure that the manifest for the sled agent exists in a location where it may
 # be automatically initialized.
-mkdir -p "$tmp/root/lib/svc/manifest/site/"
-mv pkg/manifest.xml "$tmp/root/lib/svc/manifest/site/sled-agent.xml"
+mkdir -p "$tmp_gz/root/lib/svc/manifest/site/"
+mv pkg/manifest.xml "$tmp_gz/root/lib/svc/manifest/site/sled-agent.xml"
+cd -
+# Extract the mg-ddm tarball for re-packaging into the layered GZ archive.
+pkg_dir="$tmp_gz/root/opt/oxide/mg-ddm"
+mkdir -p "$pkg_dir"
+cd "$pkg_dir"
+tar -xvfz "$tarball_src_dir/maghemite.tar"
+cd -
+
+mkdir -p /work
+cd "$tmp_gz" && tar cvfz /work/global-zone-packages.tar.gz oxide.json root
 cd -
 
+#
+# Global Zone files for for Trampoline image
+#
+
+if ! tmp_trampoline=$(mktemp -d); then
+  exit 1
+fi
+trap 'cd /; rm -rf "$tmp_trampoline"' EXIT
+
+echo '{"v":"1","t":"layer"}' > "$tmp_trampoline/oxide.json"
+
+# Extract the installinator tarball for re-packaging into the layered GZ archive.
+pkg_dir="$tmp_trampoline/root/opt/oxide/installinator"
+mkdir -p "$pkg_dir"
+cd "$pkg_dir"
+tar -xvfz "$tarball_src_dir/installinator.tar"
+# Ensure that the manifest for the installinator exists in a location where it may
+# be automatically initialized.
+mkdir -p "$tmp_trampoline/root/lib/svc/manifest/site/"
+mv pkg/manifest.xml "$tmp_trampoline/root/lib/svc/manifest/site/installinator.xml"
+cd -
 # Extract the mg-ddm tarball for re-packaging into the layered GZ archive.
-pkg_dir="$tmp/root/opt/oxide/mg-ddm"
+pkg_dir="$tmp_trampoline/root/opt/oxide/mg-ddm"
 mkdir -p "$pkg_dir"
 cd "$pkg_dir"
 tar -xvfz "$tarball_src_dir/maghemite.tar"
 cd -
 
 mkdir -p /work
-cd "$tmp" && tar cvfz /work/global-zone-packages.tar.gz oxide.json root
+cd "$tmp_trampoline" && tar cvfz /work/trampoline-global-zone-packages.tar.gz oxide.json root
```

Comment:
This tarball can be unpacked, and it'll contain:

```
oxide.json
root/opt/oxide/installinator/...
root/opt/oxide/mg-ddm/...
root/lib/svc/manifest/site/installinator.xml
```

This can be used by the image-building tools (via the `-P` flag) and it should be able to construct a recovery image which includes all these files